library/datasort.c

```diff
@@ -293,9 +303,16 @@ static char *datasort_mkdtemp(struct datasort_cfg *dcfg)
 		goto err;
 	}
 
-	if (datasort_base_get_path(dcfg->b, dcfg->bctl[0], path, PATH_MAX) != 0) {
-		EBLOB_WARNX(dcfg->log, EBLOB_LOG_ERROR, "defrag: datasort_base_get_path");
-		goto err_free_path;
+	if (for_chunks) {
+		if (datasort_chunk_get_path(dcfg->b, dcfg->bctl[0], path, PATH_MAX) != 0) {
+			EBLOB_WARNX(dcfg->log, EBLOB_LOG_ERROR, "defrag: datasort_chunk_get_path");
+			goto err_free_path;
+		}
+	} else {
+		if (datasort_base_get_path(dcfg->b, dcfg->bctl[0], path, PATH_MAX) != 0) {
```

Comment:
I recommend not adding those `for_chunks` options, but instead always use `datasort` path, which by default equals to base directory.